src/test/java/com/oltpbenchmark/benchmarks/auctionmark/TestAuctionMarkWorker.java

```diff
@@ -61,7 +60,7 @@ public void testUniqueSellers() throws Exception {
         Set<UserId> all_users = new HashSet<UserId>();
         Set<UserId> worker_users = new TreeSet<UserId>();
         Integer last_num_users = null;
-        for (Worker w : this.workers) {
+        for (var w : this.workers) {
```

Comment:
Why the change?

Reply:
TL;DR: it made the error go away 😅

`this.workers` returns a `List<Worker<? extends BenchmarkModule>>` (i.e., a wildcard for any number of classes that extend `BenchmarkModule`).
Critically, it doesn't return `List<Workers<T extends BenchmarkModule>>`.
That means that `Worker w : this.workers` is technically an invalid cast.
But so is `Worker<AuctionMarkBenchmark>` even though that's what's actually in the list, despite what the type declaration says.
The problem is that the `this.workers` value actually gets populated from an overload of `BenchmarkModule` baseclass, `AuctionMarkBenchmark`, but that means that we'd wind up with a circular type definition (where the `AuctionMarkBenchmark extends BenchmarkModule` needs to return a `Worker<AuctionMarkBenchmark>`).
Anyways, that is quite complex so I wanted to push it off to another PR (if at all) and it turns out that by simply letting java infer the type via a `var` declaration, we avoid the issue (for now).
We're still stuck with the explicit cast after that though unfortunately.